source/kraken/apply_disruption.cpp

```diff
@@ -115,24 +116,41 @@ struct apply_impacts_visitor : public boost::static_visitor<> {
     }
 };
 
-static type::ValidityPattern compute_vp(const std::vector<boost::posix_time::time_period>& periods,
+// Computes the vp corresponding to the days where base vj's are disrupted
+static type::ValidityPattern compute_base_disrupted_vp(
+        const std::vector<boost::posix_time::time_period>& disrupted_vj_periods,
         const boost::gregorian::date_period& production_period) {
     type::ValidityPattern vp{production_period.begin()}; // bitset are all initialised to 0
-    for (const auto& period: periods){
-        // we may impact vj's passed midnight
-        bg::day_iterator titr(period.begin().date());
-        for (; titr <= period.end().date(); ++titr) {
-            if (! production_period.contains(*titr)) { continue; }
-            auto day = (*titr - production_period.begin()).days();
-            vp.add(day);
-            // Why break here?
-            // There'll be a problem with VJ passes midnight.
-            // Ex:
-            // We'd like to delay a train on 1st, Jan which will pass midnight, without 'break',
-            // the impact will be applied on 1st, Jan AND 2nd, Jan. So the VJ will be de activated on 1st AND 2nd Jan,
-            // which may not be wanted
-            break;
+    for (const auto& period: disrupted_vj_periods) {
+        auto start_date = period.begin().date();
+        if (! production_period.contains(start_date)) { continue; }
+        // we may impact vj's passing midnight but all we care is start date
+        auto day = (start_date - production_period.begin()).days();
+        vp.add(day);
+    }
+    return vp;
+}
+
+// Computes the vp corresponding to the the new vj (i.e. realtime vj)
+static type::ValidityPattern compute_new_vp(
+        const nt::disruption::Impact& impact,
+        const boost::gregorian::date_period& production_period) {
+    type::ValidityPattern vp{production_period.begin()}; // bitset are all initialised to 0
+    for (const auto& period: impact.application_periods) {
+        auto start_date = period.begin().date();
+        if (! production_period.contains(start_date)
+                || impact.aux_info.stop_times.empty()) {
+            continue;
         }
+        // we may impact vj's passing midnight but all we care is start date
+        auto day = (start_date - production_period.begin()).days();
+        int day_offset = impact.aux_info.stop_times.front().arrival_time / ndtu::SECONDS_PER_DAY;
+        // cannot happen as arrival_time is unsigned, but in order to advance vj,
+        // it should be possible to have first stop_time negative
+        if (impact.aux_info.stop_times.front().arrival_time < 0) {
```

Comment:
warning here at compilation (arrival-time is unsigned): what do you prefer ?
* signed cast
* completely remove this (and we might never think of it before doing a test on forward disruption if we decide to handle them)
* other ?